Assets/Plugins/Source/Editor/ConfigEditors/ConfigEditor.cs

```diff
@@ -148,6 +224,12 @@ protected void RenderConfigFields()
                         case ConfigFieldType.Uint:
                             field.FieldInfo.SetValue(config, GUIEditorUtility.RenderInputField(field.FieldDetails, (uint)field.FieldInfo.GetValue(config), labelWidth));
                             break;
+                        case ConfigFieldType.Button:
+                            if (GUILayout.Button(field.FieldDetails.Label))
+                            {
+                                ((Action)field.FieldInfo.GetValue(config))();
```

Comment:
It surprises me that all of the other things call FieldInfo.SetValue, with a FieldInfo.Getvalue to do the thing. I can't tell if this is a problem here, or if it should be calling some kind of SetValue.

Reply:
I should not be calling a `SetValue`. The purpose of marking a field member (specifically an action) in a `Config` as a `Button` type is to inject arbitrary functionality into the UI in reaction to the user clicking the button. Because `ConfigEditor` is a template class that takes as a template parameter a `Config` type, and because it thereafter generically deals with each config - rendering the config field members in an automatic manner, it cannot know what action to take when a button is clicked, so the information about what button to render and what action to subsequently invoke must be attached to the `Config` instance.

I have updated this section of code to make that clearer by requiring the field member assigned as a `Button` be an `Action` type.